app/controllers/consent_responses_controller.rb

```diff
@@ -13,22 +13,61 @@ def create
       parent_phone: "07412 345678",
       parent_email: "test.parent@example.com",
       parent_relationship: "mother",
-      consent: "given",
       route: "website",
       health_questions: ConsentResponse::HEALTH_QUESTIONS
         .fetch(:hpv)
         .map { |question| { question: } }
     )
 
-    redirect_to action: :edit_questions
+    redirect_to action: :edit_agree
   end
 
   def update
-    @draft_consent_response.health_questions.each_with_index do |hq, index|
-      hq.merge! consent_response_health_questions_params["question_#{index}"]
+    if consent_response_agree_params.present?
+      @draft_consent_response.assign_attributes(consent_response_agree_params)
+      if @draft_consent_response.save(context: :edit_consent)
+        # Reset the reason for refusal so the user has to pick it again.
+        # Otherwise it will be pre-filled with the previous value.
+        @draft_consent_response.update! reason_for_refusal: nil
+
+        if @draft_consent_response.consent_given?
+          redirect_to action: :edit_questions
+        elsif @draft_consent_response.consent_refused?
+          redirect_to action: :edit_reason
+        else
+          redirect_to action: :edit_confirm
+        end
+      else
+        render :edit_agree
+      end
+    end
+
+    if consent_response_reason_params.present?
+      @draft_consent_response.assign_attributes(consent_response_reason_params)
+      if @draft_consent_response.save(context: :edit_reason)
+        redirect_to action: :edit_confirm
+      else
+        render :edit_reason
+      end
+    end
```

Comment:
Testing on the deployed review apps server, it seems the validations for `consent_response_agree_params` aren't working, it doesn't render anything.

Reply:
https://github.com/nhsuk/record-childrens-vaccinations/assets/15608/e679d1ce-2eb5-4145-8f98-b09b77c15827